cupy/_manipulation/basic.py

```diff
@@ -35,7 +35,7 @@ def copyto(dst, src, casting='same_kind', where=None):
         fusion._FusionVarScalar, _fusion_interface._ScalarProxy)
     if src_is_python_scalar:
         src_dtype = numpy.dtype(type(src))
-        can_cast = numpy.can_cast(src, dst.dtype, casting)
+        can_cast = numpy.can_cast(src_dtype, dst.dtype, casting)
```

Comment:
Maybe the pragmatic thing to just do this.

The more correct thing would probably be to do:
```
# When the cast is safe, we check whether the value fits (for now)
can_cast = numpy.can_cast(
    src_dtype, dst.dtype,
    casting if casting != "safe" else "same_kind
)
if casting == "safe":
    can_cast = np.array(src, dtype=dst.dtype) == src
```
You could skip the second part, but it will give you a change in behavior for NumPy 1.x and NumPy 2 currently does this anyway.  (There is no ideal solution, I think, so this is fine.)